src/native/managed/cdacreader/src/Legacy/SOSDacImpl.IXCLRDataProcess.cs

```diff
@@ -26,7 +26,56 @@ int IXCLRDataProcess.EndEnumTasks(ulong handle)
         => _legacyProcess is not null ? _legacyProcess.EndEnumTasks(handle) : HResults.E_NOTIMPL;
 
     int IXCLRDataProcess.GetTaskByOSThreadID(uint osThreadID, /*IXCLRDataTask*/ void** task)
-        => _legacyProcess is not null ? _legacyProcess.GetTaskByOSThreadID(osThreadID, task) : HResults.E_NOTIMPL;
+    {
+        // Find the thread correspending to the OS thread ID
+        Contracts.IThread contract = _target.Contracts.Thread;
+        TargetPointer thread = contract.GetThreadStoreData().FirstThread;
+        TargetPointer matchingThread = TargetPointer.Null;
+        while (thread != TargetPointer.Null)
+        {
+            Contracts.ThreadData threadData = contract.GetThreadData(thread);
+            if (threadData.OSId.Value == osThreadID)
+            {
+                matchingThread = thread;
+                break;
+            }
+
+            thread = threadData.NextThread;
+        }
+
+        if (matchingThread == TargetPointer.Null)
+            return HResults.E_INVALIDARG;
+
+        ComWrappers cw = new StrategyBasedComWrappers();
```

Comment:
Not for this PR specifically, but I've wondered in the past what this thing is and whether it's expensive